### PR TITLE
feat(datasource): rename `node` -> `node-version`

### LINKED ISSUE #20844: Rename datasource `node` to `node-version`
### Describe the proposed change(s).

Rename to be consistent with other datasources like `dotnet-version`.

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- rename `node` -> `node-version`

## Context
- Closes: https://github.com/renovatebot/renovate/issues/20844

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
@viceice Do we need to migrate `datasourceTemplate` as well?
eg.
```json
"regexManagers": [
    {
      "fileMatch": ["^main.yml$"],
      "matchStringsStrategy": "combination",
      "matchStrings": [
        "prometheus_image:\\s*\"(?<depName>.*)\"\\s*//",
        "prometheus_version:\\s*\"(?<currentValue>.*)\"\\s*//"
      ],
      "datasourceTemplate": "docker"
    }
]
```

**Review by viceice — CHANGES_REQUESTED** [review]
we need to migrate `hostType` 😱

@rarkins we haven't done it for other renames and until yet nobody complained 🤔

**Comment by RahulGautamSingh** [comment]
We don't have to migrate `datasourceTemplate` separately.

Here we can see that both `datasource` and `datasourceTemplate` value are sent to updateDependecy fn.
Note that `field` value in both cases is `datasource`.
https://github.com/renovatebot/renovate/blob/0d000aea1d3dc70eb6f0c5453b854fa519e71f46/lib/modules/manager/regex/utils.ts#L60-L76

And we migrate datasource:
https://github.com/renovatebot/renovate/blob/0d000aea1d3dc70eb6f0c5453b854fa519e71f46/lib/modules/manager/regex/utils.ts#L41

**Comment by RahulGautamSingh** [comment]
> we need to migrate hostType 😱
> @rarkins we haven't done it for other renames and until yet nobody complained 🤔

Maybe cause the affected datasources aren't used as hostType or the option itself is less used.

File: lib/config/presets/internal/regex-managers.spec.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ describe('config/presets/internal/regex-managers', () => {
           pull_request:
 
         env:
-          # renovate: datasource=node depName=node versioning=node
+          # renovate: datasource=node-version depName=node versioning=node
```
Comment (viceice) [inline lib/config/presets/internal/regex-managers.spec.ts:85]: shouldn't we leave it to validate migration?

File: lib/config/presets/internal/regex-managers.spec.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ describe('config/presets/internal/regex-managers', () => {
                   version: \${{ env.PNPM_VERSION }}
               - uses: actions/setup-node@v3
                 with:
-                  node-version: \${{ env.NODE_VERSION }}
+                  node: \${{ env.NODE_VERSION }}
```
Comment (viceice) [inline lib/config/presets/internal/regex-managers.spec.ts:101]: this change is wrong

Comment (RahulGautamSingh) [inline lib/config/presets/internal/regex-managers.spec.ts:101]:
```suggestion
                  node-version: \${{ env.NODE_VERSION }}
```

**Review by viceice — APPROVED** [review]

**Pull request merged**